src/waf.cpp

```diff
@@ -75,7 +75,7 @@ waf *waf::from_config(
 
         ddwaf::ruleset rs;
         parser::parse(ruleset, info, rs, cfg);
-        return new waf(std::move(rs), std::move(cfg));
+        return std::shared_ptr<waf>(new waf(std::move(rs), std::move(cfg)));
```

Comment:
Fine I guess, but it's a bit prone to error in the future. It would be better to encapsulate the shared_ptr creation in a static method and make the constructor of waf private. The reason being that calling `shared_from_this` without a `shared_ptr` existing is invalid (say `new waf(...)->create_context(...)`). The encapsulation of the constructor would avoid that state of affairs.